frontend/src/hooks/useDoctors.js

```diff
@@ -13,13 +16,13 @@ export default function useDoctors() {
 	});
 
 	useEffect(() => {
-		getDoctors()
+		getDoctors(token)
 			.then((doctors) => setAllDoctors(doctors))
 			.catch((error) => console.error(error.message));
-	}, [allDoctors]);
+	}, [allDoctors, token]);
 
 	const removeDoctor = (doctorId) => {
-		deleteDoctor(doctorId).then(() =>
+		deleteDoctor(doctorId, token).then(() =>
 			setAllDoctors(allDoctors.filter((doctor) => doctor.id !== doctorId))
```

Comment:
setAllDoctors will trigger the useEffect. So you are setting your doctors here, and then in the useEffect you're getting your doctors again and set them once more. Kind of redundant.
The only thing I can come up with right now would be just changing another state that you use as a trigger, so you wouldn't have to call the setDoctors twice. Don't know if that would be much cleaner

Reply:
I know, I looked at this with Matthias, but we didn't come up with a better way yet